conf/mirror.go

```diff
@@ -2,97 +2,39 @@ package conf
 
 import (
 	"fmt"
-	"net/url"
-	"os"
 	"strings"
 
 	"github.com/packethost/pkg/env"
 	"github.com/pkg/errors"
 )
 
-const (
-	defaultFacility   = "ewr1"
-	defaultMirrorPath = "/misc/tinkerbell"
-)
+const defaultFacility = "ewr1"
 
 var (
 	FacilityCode  = env.Get("FACILITY_CODE", defaultFacility)
-	mirrorURL     = mustBuildMirrorURL()
-	MirrorURL     = mirrorURL.String()
-	MirrorHost    = mirrorURL.Host
-	MirrorPath    = mirrorURL.Path
-	MirrorBase    = strings.TrimSuffix(MirrorURL, MirrorPath)
-	mirrorBaseUrl = mustBuildMirrorBaseURL()
-	MirrorBaseUrl = mirrorBaseUrl.String()
+	MirrorBaseURL = mustBuildMirrorBaseURL()
 )
 
-func buildMirrorURL() (*url.URL, error) {
-	base, err := buildMirrorBaseURL()
-	if err != nil {
-		return nil, err
-	}
-	if s, ok := os.LookupEnv("MIRROR_PATH"); ok {
-		u, err := base.Parse(s)
-		if err != nil {
-			return nil, errors.Wrapf(err, "invalid MIRROR_PATH %s", s)
-		}
-
-		return u, nil
-	}
-	u, err := base.Parse(defaultMirrorPath)
-	if err != nil {
-		return nil, errors.Wrapf(err, "invalid default mirror path %q", defaultMirrorPath)
+func mustBuildMirrorBaseURL() string {
+	s, err := buildMirrorBaseURL()
+	if s == "" {
+		panic(fmt.Sprintf(`conf: MirrorBaseURL: %q: %s\n`, s, err))
```

Comment:
this was already here from before, I've actually removed one other case of unwanted behavior. I'll rework the code though so that we don't need the conf package anymore.

Reply:
I tried re-working but then came across all the other things in https://github.com/tinkerbell/boots/blob/main/conf/config.go and stopped trying. I've made it so osie is better about this as a small step in the right direction. Getting rid of all the `must*` in conf is going to explode this PR even further.